pkg/agent/flannel/setup.go

```diff
@@ -94,8 +99,9 @@ func Run(ctx context.Context, nodeConfig *config.Node, nodes v1.NodeInterface) e
 	}
 	logrus.Info("Node CIDR assigned for: " + nodeName)
 
+	dualStack, _ := utilsnet.IsDualStackCIDRs(nodeConfig.AgentConfig.ClusterCIDRs)
```

Comment:
We are swallowing the error here because we have validated our data prior?

Reply:
Ah, assuming that `Prepare()` is invoked before `Run()` that would seem to be the case. Maybe we could cache the determination early in `Prepare()` as a side-effect (assuming that treat `nodeConfig.AgentConfig.ClusterCIDRs` as immutable). Maybe it's not worth it :man_shrugging:

Reply:
Exactly, we are already checking `IsDualStackCIDRs` before. What do you mean by caching it as a side-effect? Passing it as a value to Run()?

Reply:
> Exactly, we are already checking `IsDualStackCIDRs` before. What do you mean by caching it as a side-effect? Passing it as a value to Run()?

Yeah, I dunno. My thought was that if we make the determination the first time, and don't swallow the error  then we should save the value for later use (leaf off of `nodeConfig.AgentConfig` maybe?) so as to avoid the gratuitous reevaluation while also ignoring the error. Implementation possibly uglier and/or more error-prone than what you've got. I leave the judgement call up to you.

Reply:
Code changed ;)